src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -1043,5 +1154,22 @@ TEST(writer_reader, writer_resize_block) {
     }
 }
 
+TEST(writer_reader, source_block_max) {
+    check_writer_source_block_max(rs_source_composer, rs_repair_composer, ReedSolomon8m,
```

Comment:
I suggest to use the same approach as in encoder-decoder tests:

* create a cycle over all codec types in every test;
* inline test code (check_writer_source_block_max) in the test itself.

Pros:

* will automatically work with new codec if we add them;
* a single test-case code is not dispersed in different parts of the file.

Reply:
Done